cms/envs/common.py

```diff
@@ -718,6 +721,12 @@
 
     # This must be last so that it runs first in the process_response chain
     'openedx.core.djangoapps.site_configuration.middleware.SessionCookieDomainOverrideMiddleware',
+
+    # CORS and CSRF
+    'corsheaders.middleware.CorsMiddleware',
```

Comment:
If we don't configure CORS on studio, is it the case the API calls from course-authoring MFE won't succeed because of cross domain request gets rejected by studio APIs?

Reply:
Yes. Without this, the necessary CORS headers (i.e. `Access-Control-Allow-Origin`) are not included in Studio's response to a preflight request that is made by the MFE, leading the browser refusing to make a GET request (or any request that required a preflight request, like a POST). You will see this error in the developer console.

```Access to XMLHttpRequest at 'http://localhost:18010/test/exam_settings' from origin 'http://localhost:2001' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.```

In order to correctly use the[ corsheaders library](https://pypi.org/project/django-cors-headers/, which supplies the necessary headers, we need to include the required middleware and add `corsheaders` to `INSTALLED_APPS`. I copied these settings from `lms/envs/common.py`.

However, I'm seeing a lot of discussion on this topic in [this PR](https://github.com/edx/edx-platform/pull/24454), so I'm going to read through that PR. I'm not completely comfortable merging these Django settings changes without knowing the implications.

I could remove these changes from this pull request temporarily in order to wait for that other PR to land (and set these settings during development).